src/main/java/bio/terra/service/filedata/azure/tables/TableFileDao.java

```diff
@@ -0,0 +1,141 @@
+
+  @Autowired
+  TableFileDao(
+      FireStoreUtils fireStoreUtils, @Qualifier("performanceThreadpool") ExecutorService executor) {
+    this.fireStoreUtils = fireStoreUtils;
+    this.executor = executor;
+  }
+
+  public void createFileMetadata(TableServiceClient tableServiceClient, FireStoreFile newFile) {
+    tableServiceClient.createTableIfNotExists(TABLE_NAME);
+    TableClient tableClient = tableServiceClient.getTableClient(TABLE_NAME);
+    TableEntity entity =
+        new TableEntity(PARTITION_KEY, newFile.getFileId())
+            .addProperty("fileId", newFile.getFileId())
+            .addProperty("mimeType", newFile.getMimeType())
+            .addProperty("description", newFile.getDescription())
+            .addProperty("bucketResourceId", newFile.getBucketResourceId())
+            .addProperty("loadTag", newFile.getLoadTag())
+            .addProperty("fileCreatedDate", newFile.getFileCreatedDate())
+            .addProperty("gspath", newFile.getGspath())
+            .addProperty("checksumCrc32c", newFile.getChecksumCrc32c())
+            .addProperty("checksumMd5", newFile.getChecksumMd5())
+            .addProperty("size", newFile.getSize());
+    logger.info("creating file metadata for fileId: " + newFile.getFileId());
+    tableClient.createEntity(entity);
+  }
+
+  public boolean deleteFileMetadata(TableServiceClient tableServiceClient, String fileId) {
+    TableClient tableClient = tableServiceClient.getTableClient(TABLE_NAME);
+    TableEntity entity = tableClient.getEntity(PARTITION_KEY, fileId);
+    if (entity == null) {
+      return false;
+    }
+    logger.info("deleting file metadata for fileId " + fileId);
+    tableClient.deleteEntity(PARTITION_KEY, fileId);
+    return true;
+  }
+
+  public FireStoreFile retrieveFileMetadata(TableServiceClient tableServiceClient, String fileId) {
+    TableClient tableClient = tableServiceClient.getTableClient(TABLE_NAME);
+    TableEntity entity = tableClient.getEntity(PARTITION_KEY, fileId);
+    return FireStoreFile.fromTableEntity(entity);
+  }
+
+  /**
+   * Retrieve metadata from a list of directory entries.
+   *
+   * @param tableServiceClient An Azure table service client
+   * @param directoryEntries List of objects to retrieve metadata for
+   * @return A list of metadata object for the specified files. Note: the order of the list matches
+   *     with the order of the input list objects
+   */
+  List<FireStoreFile> batchRetrieveFileMetadata(
+      TableServiceClient tableServiceClient, List<FireStoreDirectoryEntry> directoryEntries) {
+    List<FireStoreFile> files = new ArrayList<>();
+    for (FireStoreDirectoryEntry f : directoryEntries) {
+      FireStoreFile fsFile = retrieveFileMetadata(tableServiceClient, f.getFileId());
+      if (fsFile == null) {
+        throw new FileSystemCorruptException("Directory entry refers to non-existent file");
+      }
+      files.add(fsFile);
+    }
+    return files;
+  }
+
+  <V> void scanTableObjects(
```

Comment:
I'm curious: do we even need this batching logic?  I thought that we were doing all of this because of firestore limitations?

Reply:
I'm not entirely sure...I know we can delete the whole table, but do we need to delete the associated Azure files in batches?

Reply:
Interested in this, as it has implications for the load history table

Reply:
I took out the batching here so it's no longer re-using the firestore utils `batchOperation` method